rfcs/0005-moxy-carousel.md

```diff
@@ -0,0 +1,195 @@
+    <div className={ classNames(styles.slide, className) }>{title}</div>
+);
+
+const slidesData = [
+    { title: 'item 1', className: styles.firstSlide },
+    { title: 'item 2', className: styles.secondSlide },
+    { title: 'item 3', className: styles.thirdSlide },
+    { title: 'item 4', className: styles.forthSlide },
+];
+
+const Home = () => (
+    <Carousel
+        swipeable={ false }
+        draggable={ false }
+        showDots
+        responsive={ responsive }
+        ssr // Means to render carousel on server-side.
+        infinite
+        autoplay={ false }
+        autoPlaySpeed={ 1000 }
+        keyBoardControl
+        customTransition="all .5"
+        transitionDuration={ 500 }
+        containerClass={ styles.containerClass }
+        removeArrowOnDeviceType={ ['tablet', 'mobile'] }
+        dotListClass="custom-dot-list-style"
+        centerMode
+        itemClass={ styles.itemClass }
+        activeItemClass={ styles.activeItemClass }>
+        {slidesData.map((data) => (
+            <Slide key={ data.title } { ...data } />
+        ))}
+    </Carousel>
+);
+
+export default Home;
+```
+
+- `Home.module.css`
+
+```css
+@import "../../shared/styles/variables";
+
+.home {
+    width: var(--home-width);
+    height: var(--home-height);
+    background-color: var(--color-white);
+}
+
+.containerClass {
+    margin-top: 200px;
+
+    & .itemClass {
+        display: flex;
+        justify-content: center;
+
+        opacity: 0.3;
+
+        & .slide {
+            width: 100%;
+            height: 200px;
+
+            display: flex;
+            justify-content: center;
+            align-items: center;
+
+            color: whitesmoke;
+            font-weight: 700;
+            text-transform: uppercase;
+        }
+
+        & .firstSlide {
+            background-color: hotpink;
+        }
+
+        & .secondSlide {
+            background-color: purple;
+        }
+
+        & .thirdSlide {
+            background-color: darkcyan;
+        }
+
+        & .forthSlide {
+            background-color: darksalmon;
+        }
+    }
+
+    & .activeItemClass {
+        opacity: 1;
+    }
+}
+```
+
+# Motivation
+
+Everytime we need to implement a carousel/slider we are dependant on external libs, usually these libs meet the basic requirements, but eventually they will become problematic for more extended use cases.
+
+It would be of great value to have our own implemented carousel/slider to prevent extended overhead time when implementing the wrapper for the external module.
+
+# Detailed design
+
+The original `react-multi-carousel` was lacking a way to style the active item(s) without overriding the package's CSS, so I added a new prop called `activeItemClass` that accepts a `class`.
+
+More changes and/or additions will come after discussion about our needs.
+
+# Drawbacks
+
+- Depending on how many new features we will add, this will drastically diverge from upstream and thus it will require more effort to maintain;
```

Comment:
We should make a PR to the react-multi-carousel package with this feature nonetheless. If they don't accept, we keep using this fork.

Reply:
Sure, will do it!

Reply:
Done! https://github.com/YIZHUANG/react-multi-carousel/pull/125